reporthandling/results/v1/resourcesresults/datastructures.go

```diff
@@ -27,6 +29,7 @@ type ResourceAssociatedRule struct {
 	ControlConfigurations map[string][]string                `json:"controlConfigurations,omitempty"`
 	Name                  string                             `json:"name"`
 	Status                apis.ScanningStatus                `json:"status"`
+	SubStatus             apis.ScanningSubStatus             `json:"subStatus"`
```

Comment:
We should consider using the object StatusInfo and have all status-related info handled there